aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -107,36 +107,59 @@ Tensor pinverse(const Tensor& self, double rcond) {
   return at::matmul(V.conj() * S_pseudoinv.unsqueeze(-2), U.transpose(-2, -1).conj());
 }
 
-static inline Tensor _matrix_rank_helper(const Tensor& self, bool symmetric) {
+Tensor linalg_matrix_rank(const Tensor& self, optional<double> tol, bool hermitian) {
+  // Matrices or batch of matrices are allowed
+  TORCH_CHECK(self.dim() >= 2, "linalg_matrix_rank: Expected as input a matrix or a batch of matrices, but got a tensor of size: ", self.sizes());
+
+  // NumPy doesn't take into account possible input with no elements and it errors on max not defined for this case
+  // Let's output 0 for this case, since that kind of matrices have zero number of non-zero rows, hence rank is 0.
+  if (self.numel() == 0) {
+    auto result_shape = self.sizes().vec();
+    result_shape.pop_back();
+    result_shape.pop_back();  // result's shape is equal to self.shape[0:-2]
```

Comment:
Can you elaborate on this choice for the returned shape and expand the comment to describe why this result is this shape

Reply:
I expanded this comment.
`matrix_rank` assigns for each matrix in the batch a scalar, so the resulting shape is (in Python) `self.shape[0:self.ndim-2]`, the last two dimensions of `self` are removed. For single matrix input, this gives an empty tuple, or in C++ empty vector.